### PR TITLE
Fix node.js v14 support

### LINKED ISSUE #641: Issues running jazzer in node v14
The `engine` in package.json claims that nodejs version `>=14` is supported/required. But when using the most recent version of Node 14 (as part of a test) I get the following output:

```
     TypeError: Object.hasOwn is not a function
       at node_modules/@jazzer.js/core/options.ts:207:15
           at Array.forEach (<anonymous>)
       at mergeOptions (node_modules/@jazzer.js/core/options.ts:205:31)
       at node_modules/@jazzer.js/core/options.ts:179:11
           at Array.reduce (<anonymous>)
       at buildOptions (node_modules/@jazzer.js/core/options.ts:176:4)
       at loadConfig (node_modules/@jazzer.js/jest-runner/config.ts:52:21)
```

https://github.com/xmldom/xmldom/actions/runs/6452742416/job/17515178816#step:7:64

The GitHub action workflow that is used for testing in the jazzer.js repository is using node v16.

Should the value in the engines field be bumped or are you still trying to support node v14?

### PR DESCRIPTION
**Pull request by bertschneider**
Replace two functions introduced in v16 to still support v14. Supporting this old version is probably a big stretch in the JavaScript ecosystem, though.

To verify ECMAScript compatibility the TypeScript target version is now set to ES2020 as that is mostly supported by node.js v14. This only catches invalid ECMAScript features, and would have done so in case of the two fixed usages, but not unsupported node.js APIs.

Sadly, available eslint plugins verifying correct node.js API version usage seem quite out of date.

Closes #641

### CODE REVIEW

**Comment by br-lewis** [comment]
I know the ticket says otherwise but if we can add tests for 14 relatively simply, I think we should so we don't get any more surprises as long we want to support 14.

**Comment by bertschneider** [comment]
I looked into verifying correct node.js and ECMAScript usage via an `eslint` plugin, but the ones I found are quite out of date. Setting the TypeScript target version to `ES2020` seems like an easy solution to verify correct ECMAScript usage, and would have caught the problem. 

[This table](https://node.green/#ES2020) has a nice overview of ECMAScript versions and their node.js support. Using `ES2020` should be fine for now.

**Comment by bertschneider** [comment]
In the last commit I tried to execute the examples with v14, but that does not seem to work, as npm 7 is required to support workspaces during build time. npm can not distinguish between build and runtime engine versions.

npm 7 is included in node.js v15. Using that version for the examples should work, but is it worth the hassle? WDYT?

**Comment by oetr** [comment]
Would it make sense to only add some end-to-end tests for all but one node version (say v18, or v16)? 50m to test one node version, x4 = about 4 hours. With one flaky test, and we will need several days to merge a PR.

Can the e2e tests use Jazzer.js built by a different node version?

**Comment by bertschneider** [comment]
As discussed, tests and examples are now executed with all LTS versions on Linux and the end-to-end tests also with v14.

File: package.json [hunk 1]
```diff
@@ -22,12 +22,15 @@
...
 		"test:default": "npm run test:jest",
-		"test:linux:darwin": "npm run test:jest && cd examples && sh ../scripts/run_all.sh dryRun && cd ../tests && sh ../scripts/run_all.sh fuzz",
-		"test:win32": "npm run test:jest && cd examples && ..\\scripts\\run_all.bat dryRun && cd ..\\tests && ..\\scripts\\run_all.bat fuzz",
-		"test:watch": "jest --watch",
+		"test:linux:darwin": "npm run test:jest && cd tests && sh ../scripts/run_all.sh fuzz",
+		"test:win32": "npm run test:jest && cd tests && ..\\scripts\\run_all.bat fuzz",
+		"test:jest": "jest && npm run test --ws --if-present",
+		"test:jest:coverage": "jest --coverage",
+		"test:jest:watch": "jest --watch",
+		"example": "run-script-os",
```
Comment (br-lewis) [inline package.json:31]: Weren't the examples already covered under `npm run test`?

**Review by br-lewis — COMMENTED** [review]
I think the readme would be a good spot to mention our node version support. I would be inclined to say it should go towards the bottom since it isn't immediately relevant when you're trying to figure out what the project is and does. We do have architecture support at the very top so it makes sense to have it there though maybe we should move that further down as well.

File: package.json (hunk 1 above)
Comment (bertschneider) [reply package.json:31]: Yes, I extracted them out to execute only the examples with a different node.js version, but reverted that part back. Still, this should reduce a few minutes from the overall runtime as es examples are now executed in parallel.

**Comment by bertschneider** [comment]
Moved the supported architecture section further down and added a remark about supported Node.js versions.

**Pull request merged**